src/ui/map.js

```diff
@@ -2664,13 +2671,16 @@ class Map extends Camera {
             if (err) {
                 // throwing an error here will cause the callback to be called again unnecessarily
                 if (err.message === AUTH_ERR_MSG || err.status === 401) {
-                    console.error('Error: A valid Mapbox access token is required to use Mapbox GL JS. To create an account or a new access token, visit https://account.mapbox.com/');
-                    browser.setErrorState();
                     const gl = this.painter.context.gl;
+                    storeAuthState(gl, false);
```

Comment:
One question about storeAuthState/removeAuthState, how do we ensure storeAuthState always followed by removeAuthState? If this is even possible in practice, what would be the behavior in that case?

Reply:
`removeAuthState` has to be called only when the map is removed. I don't think we can gurantee it other than ensuring we never deleting the line in `Map#remove`

Reply:
Should we go over all stored states and assert if anything is leftover?